### ISSUE TITLE
yq --sort-keys options, like jq

### ISSUE THREAD

**Issue by robbat2** [mikefarah/yq#2087]
**Please describe your feature request.**
TL;DR: Please add a `yq --sort-keys` option, like `jq --sort-keys`.

`jq` has a `--sort-keys` option, that sorts the keys in the output consistently.

I use it in shell commands like:
```
diff -NuarwbB <(jq --sort-keys $EXPR file-A) <(jq --sort-keys $EXPR file-B)
```
To be able to see the differences quickly regardless of any formatting or ordering of hashes.

It should sort all hashes by the lexical order of their keys; for the entire file, regardless of depth.

This is available via `sort_keys(..)` as documented in the manpage, but how about the command-line option for ease of use, esp. with other complex expressions.

**Describe the solution you'd like**
If we have data1.yml like:
(please keep to around 10 lines )
```yaml
hash: 
  keyZ: Zambia
  keyA: Australia
```

And we run a command:

```bash
yq -P --sort-keys . 
```

it should output

```yaml
hash:
  keyA: Australia
  keyZ: Zambia
```

**Describe alternatives you've considered**
Existing workarounds:
- Convert to json, sort in JQ, convert back to YAML. This loses comments, anchors, references.
- `yq $EXPR file-A | yq -P 'sort_keys(..)'` => extra exec, sometimes forget about it

**Comment by mat813** [comment]
Ah, yes, I wanted that today, I had to resort going through jq via json with:

```
yq -o json input.yml | jq -S | yq -p json -o yaml >! sorted.yml
```

I tried to use sort_keys but it only sorts one level of keys, and here I am trying to compare `.gitlab-ci.yml` files, and those have multiple levels of objects.

**Comment by cs224** [comment]
+1

**Comment by AiyionPrime** [comment]
> [...]
> I tried to use sort_keys but it only sorts one level of keys, and here I am trying to compare `.gitlab-ci.yml` files, and those have multiple levels of objects.

@mat813 IIRC `sort_keys()` does multiple levels, if you invoke it like this: `sort_keys(..)`
